src/transformState.js

```diff
@@ -4,8 +4,28 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
   // write code here
+  for (const action in actions) {
+    switch (actions[action].type) {
+      case 'addProperties':
+        Object.assign(state, { ...actions[action].extraData });
+        break;
+      case 'removeProperties':
+        for (const keyToRemove in actions[action].keysToRemove) {
+          delete state[actions[action].keysToRemove[keyToRemove]];
+        }
```

Comment:
this loop is really complicated. after using a `for of` loop above -  you can make this one mutch simpler